wandb/sdk/artifacts/artifact.py

```diff
@@ -532,6 +532,27 @@ def type(self) -> str:
         """The artifact's type. Common types include `dataset` or `model`."""
         return self._type
 
+    @property
+    def url(self):
+        """
+        Constructs the URL of the artifact.
+
+        Returns:
+            str: The URL of the artifact.
+        """
+        if self.collection.is_sequence():
```

Comment:
**suggestion:** I would suggest splitting this up into 3 internal helper methods for readability

Reply:
Done!